graphqlapi/src/test/java/io/stargate/graphql/schema/fetchers/dml/QueryFetcherUdtsTest.java

```diff
@@ -13,12 +17,24 @@ public Keyspace getKeyspace() {
     return SampleKeyspaces.UDTS;
   }
 
-  @Test
-  @DisplayName("Should format UDT value into query")
-  public void udtTest() {
-    String graphQlQuery =
-        "query { testTable(value: { a: { b: { i: 1 } } }) { values { a{b{i}} } } }";
-    String expectedCqlQuery = "SELECT a FROM udts.test_table WHERE a={\"b\":{\"i\":1}}";
-    assertSuccess(graphQlQuery, expectedCqlQuery);
+  @ParameterizedTest
+  @MethodSource("successfulQueries")
+  @DisplayName("Should execute GraphQL with UDTs and generate expected CQL query")
+  public void udtTest(String graphQlQuery, String expectedCqlQuery) {
+    assertSuccess(String.format("query { %s }", graphQlQuery), expectedCqlQuery);
+  }
+
+  public static Arguments[] successfulQueries() {
+    return new Arguments[] {
+      arguments(
+          "testTable(value: { a: { b: {i:1} } }) { values { a{b{i}} } }",
+          "SELECT a FROM udts.test_table WHERE a={\"b\":{\"i\":1}}"),
+      arguments(
+          "testTable(filter: { a: {eq: { b: {i:1} } } }) { values { a{b{i}} } }",
```

Comment:
I had made a mistake here: `a` was a UDT instance directly, in fact we need a new `AUdtFilterInput` that defines all the comparison operators as fields: eq, gt, etc.